src/Extensions.Hosting/HostBuilderExtensions.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Diagnostics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Json;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+using Serilog;
+
+namespace Bit.Extensions.Hosting;
+
+public static class HostBuilderExtensions
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public static IHostBuilder UseBitwardenDefaults(this IHostBuilder hostBuilder, Action<BitwardenHostOptions>? configure = null)
+    {
+        // We could default to not including logging in development environments like we currently do.
+        var bitwardenHostOptions = new BitwardenHostOptions();
+        configure?.Invoke(bitwardenHostOptions);
+        return hostBuilder.UseBitwardenDefaults(bitwardenHostOptions);
+    }
+
+    public static IHostBuilder UseBitwardenDefaults(this IHostBuilder hostBuilder, BitwardenHostOptions bitwardenHostOptions)
+    {
+        hostBuilder.ConfigureServices((context, services) =>
+        {
+            services.AddOptions<GlobalSettingsBase>()
+                .Configure<IConfiguration>((options, config) =>
+                {
+                    options.SelfHosted = config.GetValue("globalSettings:selfHosted", false);
+                });
+        });
+
+        hostBuilder.ConfigureAppConfiguration((context, builder) =>
+        {
+            if (context.Configuration.GetValue("globalSettings:selfHosted", false))
+            {
+                // Current ordering of Configuration:
```

Comment:
😨 ... this is the best .NET can offer? Can we move away from `appsettings.SelfHosted.json`?

Reply:
Yes we probably can but will take a little planning.

Reply:
@withinfocus - Could you provide an example of what your version of better looks like?